zookeeper-server/src/main/java/org/apache/zookeeper/server/watch/WatcherCleaner.java

```diff
@@ -106,6 +107,7 @@ public void addDeadWatcher(int watcherBit) {
                 synchronized(processingCompletedEvent) {
                     processingCompletedEvent.wait(100);
                 }
+                ServerMetrics.ADD_DEAD_WATCHER_STALL_TIME.add(100);
```

Comment:
Is 100 right?
Are we supposing that the abow *wait* always timeouts?
Is It worth to track the exact time?

Reply:
Good catch @eolivelli . Accuracy of this metric is <100ms now, which we can improve by measuring the exact time elapsed. I'm not sure either if it's worth the extra logic.

Reply:
@anmolnar 

>  I'm not sure either if it's worth the extra logic

me too

Reply:
@jhuan31 what do you think ?

Reply:
Nice catch, I think we should measure the exact time used when waiting for the signal, @jhuan31 can you help update this.

Reply:
Sorry about the late response. I will fix it. Thank you @eolivelli  @anmolnar  @lvfangmin